### PR TITLE
Remove CIMCORE exporter.

### PR DESCRIPTION
**Pull request by mint-thompson**
Exporting CIMCORE files is no longer a prerequisite for exporting the ModelDoc. CIMCORE files can no longer be exported. The command line option to skip the CIMCORE export is removed.

It might also be useful to update shr-json-javadoc such that variables are no longer named "cimcore", but that is not required to make this change work correctly.

### CODE REVIEW

**Comment by cmoesel** [comment]
Does this actually work with no changes to shr-json-javadoc?  I don't see any branches on shr-json-javadoc to accomodate the shr-models rather than the cimcore.  Are they *that* compatible?

**Comment by kjmahalingam** [comment]
I ran this without any changes, using `master` on all branches, other than this one. It worked fine, with no errors, and the expected number of warnings. I don't know what else I should do to verify.

@cmoesel have you tested this yet, or were you waiting on an answer to your question? Assuming that you see the model doc looking the way it should...perhaps it was that compatible.

**Comment by kjmahalingam** [comment]
Works without issue:

```
------------------------------------------------------------
Elapsed time: 81.792s
0 errors
12 warnings (shr-text-input, shr-fhir-export)
------------------------------------------------------------
MM232274-PC:shr-cli kmahalingam$
```

I verified that I had the `includeModelDoc` flag in the config set to `true`.

@cmoesel Did you test this yet or were you waiting on an answer? Assuming that you saw the model doc looking like it should, then perhaps this was as clear as it looks.

**Comment by cmoesel** [comment]
I haven't actually tested it.  I was just surprised that there were no corresponding model-doc changes.  Maybe it *does* work as-is, which would be amazing.  I'll test it later (when I get a chance).

**Comment by cmoesel** [comment]
I've confirmed that this works, but... the reason it works is that we're not really getting rid of CIMCORE this way.  We're not producing it directly in the CLI, but it's still actually getting produced and used in the `shr-json-javadoc` itself.  I'll try to explain...

Abhijay implemented CIMCORE export by adding `toJSON()` functions to all of the `shr-models` classes (e.g., `DataElement`, `TypeConstraint`, `ValueSetConstraint`, etc.).  For example:
https://github.com/standardhealth/shr-models/blob/master/lib/models.js#L635-L649

_Before this PR_, the CLI would call `toJSON()` on all of the data elements before passing them to the `shr-json-javadoc`.  For example:
```javascript
let out = Object.assign({ 'fileType': 'DataElement' }, de.toJSON());
cimcoreSpecifications.dataElements.push(out);
```
So... this is how _CIMCORE_ was being passed to the `shr-json-javadoc`.

_In this PR_ we are passing in the `shr-models` instances of data elements more directly, *without* calling `toJSON()` on them.  For example:
```
  dataElements: expSpecifications.dataElements.all,
```
So, we're _not_ passing in CIMCORE.  Great!

But... now in the `shr-json-javadoc` module, we find this code:
```javascript
for (const de of cimcore.dataElements) {
  const deClone = JSON.parse(JSON.stringify(de));
  this.elements.add(deClone);
  // ... snipping out stuff not relevant to this discussion
}
```
It's easy to miss, _but_ `JSON.stringify` actually checks its argument to see if it has a `toJSON` function on it, and if it does, it actually invokes `toJSON()` on the argument and stringifies the result! See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON()_behavior

So... while we've removed CIMCORE from the CLI, the `shr-json-javadoc` still actually relies on the CIMCORE produced by the `toJSON()` functions.  If the purpose of this task is merely to stop exporting CIMCORE in the CLI, then we're good.  But if the intent is to actually _deprecate_ CIMCORE, then there will need to be changes to `shr-json-javadoc` to use the `shr-models` classes directly without stringifying them.  We can (and should) also talk about whether we want to leave the `toJSON()` functions on the model classes as-is, modify them to be simpler, or remove them entirely.

**Comment by cmoesel** [comment]
BTW -- we *do* have a goal of actually _deprecating_ CIMCORE, so my last paragraph isn't really intended to ask if we should deprecate CIMCORE, but rather if we mean to deprecate CIMCORE _now_.  Since I wasn't at the sprint planning, I don't know what was discussed.

**Comment by mint-thompson** [comment]
This pull request has been updated with additional changes. Furthermore, two other pull requests have been created on shr-json-javadoc and shr-models that should accompany this pull request:
* https://github.com/standardhealth/shr-json-javadoc/pull/27
* https://github.com/standardhealth/shr-models/pull/48

File: app.js [hunk 1]
```diff
@@ -408,24 +296,22 @@ if (doJSONSchema) {
...
-      logger.fatal({ errorText: error.stack }, '15010');
-      failedExports.push('shr-model-doc');
+      const igJavadocResults = shrJDE.compileJavadoc(modelDocConfig, hierarchyPath, true);
+      shrJDE.exportToPath(igJavadocResults, fhirPath);
     }
-  } else {
-    // 15011, 'CIMCORE is required for generating Model Doc. Skipping Model Docs export.', 'Do not skip CIMCORE if Model Doc should be generated', 'errorNumber'
-    logger.fatal('15011');
+  } catch (error)  {
+    logger.fatal('Failure in Model Doc export. Aborting with error message: %s', error);
```
Comment (cmoesel) [inline app.js]: Note that there is a new approach to error handling.  You need to create a new message (w/ a unique number) in the errorMessages.txt file and just log the error by number (e.g., `logger.fatal('15999');`).

Comment (mint-thompson) [reply app.js]: Thank you! It's now using error message 15010, which already existed and had the correct contents.

**Pull request merged**